test/hooks-async.js

```diff
@@ -126,6 +126,78 @@ function asyncHookTest (t) {
       t.strictEqual(res.headers['content-length'], '22')
     })
   })
+
+  test('onRequest hooks should be able to block a request', t => {
```

Comment:
Maybe these cases should be worded: "should be able to terminate a request"? The word "block" seems to imply something more sinister to me.